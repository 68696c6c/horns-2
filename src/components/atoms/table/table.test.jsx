/* global describe, it, expect */
import React from 'react'

import { render } from 'setup-test'

import TableScroll from '.'

describe('Table', () => {
  it('should render as default', () => {
    const { container } = render(
      <TableScroll
        minWidth="1200px"
        rowData={[
          { id: '1', name: 'first', email: 'first@example.com' },
          { id: '2', name: 'second', email: 'second@example.com' },
        ]}
      />
    )
    expect(container).toMatchSnapshot()
  })
  it('should render with static content', () => {
    const { container } = render(
      <TableScroll minWidth="400px">
        <thead>
          <tr>
            <th>zero one</th>
            <th>zero two</th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td>one one</td>
            <td>one two</td>
          </tr>
          <tr>
            <td>two one</td>
            <td>two two</td>
          </tr>
        </tbody>
      </TableScroll>
    )
    expect(container).toMatchSnapshot()
  })
})
