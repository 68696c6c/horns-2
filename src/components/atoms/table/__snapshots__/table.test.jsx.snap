// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table should render as default 1`] = `
.emotion-1 {
  width: 100%;
  height: auto;
  overflow-x: scroll;
}

.emotion-0 {
  width: 100%;
  border-collapse: collapse;
  min-width: 1200px;
}

.emotion-0 td,
.emotion-0 th {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 8px;
  padding-right: 8px;
}

.emotion-0 th {
  font-family: Helvetica,sans-serif;
  font-size: medium;
  font-style: normal;
  font-weight: 700;
  line-height: 1em;
}

.emotion-0 td {
  font-family: Helvetica,sans-serif;
  font-size: medium;
  font-style: normal;
  font-weight: 400;
  line-height: 1em;
}

.emotion-0 tbody tr:nth-child(odd),
.emotion-0 tr:nth-child(odd):not(:nth-child(1)) {
  background: rgb(235,235,235);
  color: rgb(1,1,1);
  border-color: rgb(216,216,216);
}

<div>
  <div
    class="emotion-1"
    height="auto"
    x="scroll"
  >
    <table
      class="emotion-0"
    >
      <thead>
        <tr>
          <th>
            id
          </th>
          <th>
            name
          </th>
          <th>
            email
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>
            1
          </td>
          <td>
            first
          </td>
          <td>
            first@example.com
          </td>
        </tr>
        <tr>
          <td>
            2
          </td>
          <td>
            second
          </td>
          <td>
            second@example.com
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;

exports[`Table should render with static content 1`] = `
.emotion-1 {
  width: 100%;
  height: auto;
  overflow-x: scroll;
}

.emotion-0 {
  width: 100%;
  border-collapse: collapse;
  min-width: 400px;
}

.emotion-0 td,
.emotion-0 th {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 8px;
  padding-right: 8px;
}

.emotion-0 th {
  font-family: Helvetica,sans-serif;
  font-size: medium;
  font-style: normal;
  font-weight: 700;
  line-height: 1em;
}

.emotion-0 td {
  font-family: Helvetica,sans-serif;
  font-size: medium;
  font-style: normal;
  font-weight: 400;
  line-height: 1em;
}

.emotion-0 tbody tr:nth-child(odd),
.emotion-0 tr:nth-child(odd):not(:nth-child(1)) {
  background: rgb(235,235,235);
  color: rgb(1,1,1);
  border-color: rgb(216,216,216);
}

<div>
  <div
    class="emotion-1"
    height="auto"
    x="scroll"
  >
    <table
      class="emotion-0"
    >
      <thead>
        <tr>
          <th>
            zero one
          </th>
          <th>
            zero two
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>
            one one
          </td>
          <td>
            one two
          </td>
        </tr>
        <tr>
          <td>
            two one
          </td>
          <td>
            two two
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
